sources/packages/web/src/store/modules/student/student.ts

```diff
@@ -0,0 +1,53 @@
+
+export class StudentState {
+
+     profile!: StudentProfile;
+}
+
+export class StudentProfile {
+    name?: string;
+    age?: string;
+    birthdate?: string;
+    displayName?: string;
+    email?: string;
+    emailVerified?: string;
+    familyName?: string;
+    gender?: string;
+    givenName?: string; 
+    middleName?:string;
+}
+
+const initialState: StudentState = {
+    profile: new StudentProfile()
+}
+
+export const student = {
+    namespaced: true,
+    state : initialState,
+   
+    mutations: {
+        SET_STUDENT_PROFILE_DATA(state:StudentState, parsedToken:any){
+            //console.log(`mutation p${parsedToken.age}`)                        
+            state.profile.name = parsedToken.name;
+            state.profile.age = parsedToken.age;
+            state.profile.birthdate = parsedToken.birthdate;
+            state.profile.displayName = parsedToken.displayName;
+            state.profile.email = parsedToken.email;
+            state.profile.emailVerified = parsedToken.email_verified;
+            state.profile.familyName = parsedToken.family_name;
+            state.profile.gender = parsedToken.gender;
+            state.profile.givenName = parsedToken.given_name;
+            state.profile.middleName=parsedToken.middle_name;
```

Comment:
I believe that we are not receiving the middle_name from BCSC and I also believe that this field will be always blank.
From BCSC with receive given_names and given_name. I believe that the only way to get the middle name would be subtracting the given_names from given_name.
For instance:
Considering the given_names as "SS4BPS Benjamin"
Considering the given_name as "SS4BPS"
the middle name would be "Benjamin".
@abschwenker, could you please confirm if it is still needed?

Reply:
I will have an answer today if we need to parse out the middle name

Reply:
Removed the middlename as discussed with schwenker

Reply:
@abschwenker, can we have some comments on the story to support this change?